### DISCUSSION TITLE
[Regression] Cancelling a commit now erases the commit message, so one has to create it anew

### DISCUSSION THREAD

**Discussion by Olf0** [community/community#55880] [Repositories]
### Select Topic Area

Bug

### Body

Now cancelling a commit erases the commit message, so one has to create it anew, when finally deciding to commit.  For a [detailed description](https://github.com/orgs/community/discussions/55880#discussioncomment-5953578), see below.

Before (for years) the old commit message was retained, so one could reuse or adapt it.  Nothing prevented one from altering it to any arbitrary extent, so this constitutes a regression without any value.

Please revert this change, which created an anti-feature!

**Comment by samuelss1102** [comment]
I can guide you on how to revert a commit in a version control system like Git, assuming that's the context of your question. If you want to retain the commit message after canceling a commit, you may consider using the git reset command with the --soft flag, which undoes the commit but leaves your changes and the commit message. Here's an example:

`git reset --soft HEAD~1`

This command moves the HEAD back by one commit (the ~1 part) but leaves the changes in your working directory (the --soft part). After running it, you can commit again, and the message from the undone commit will be the default message in your text editor.

It's always a good idea to check with the specific version control system or software you're using for the most accurate and up-to-date information. I hope this helps.

**Reply by Olf0** [reply]
> I can guide you on how to revert a commit in a version control system like Git, assuming that's the context of your question. 

Sorry, this completely misses the point.  But I understand, that my initial description was too terse and lacked sufficient context.  Hence I use the chance top provide it here:

This bug / regression report is solely about GitHub's web-frontend, when one made a change to a file using the web editor, hit the "Commit changes..." button, entered as commit message in the newly opened window and then hit its "Cancel" button, so one is back in the web editor.  No commit has been submitted to Git at this point.

When one hits the "Commit changes..." button again (likely after additionally altering something via the web editor), before the previously entered commit message was shown (i.e., it was retained), now the commit message field is empty each time this window opens.  This is the regression I am trying to address.
